### ISSUE TITLE
Would scope_extensions handle this scenario?

### ISSUE THREAD

**Issue by JudahGabriel** [WICG/beforeinstallprompt#46]
A team building a large productivity PWA is encountering the following issue. Will `scope_extensions` support it?

1. User installs PWA at home.contoso.com
2. home.contoso.com asks user to sign in
3. Upon sign-in, the user is redirected to mycompany.contoso.com
4. PWA sends a push notification to the user
5. User clicks the push notification, which navigates to mycompany.contoso.com
6. BUG: clicking the push notification opens in a new browser tab, rather than in the installed PWA

That last step is the issue. They want users who click that push notification to open in the installed PWA. Instead, today, it's opening in a new browser tab.

Will `scope_extensions` support this scenario?

**Comment by alancutter** [comment]
Yes by virtue of extending the app's scope to include that other origin.

**Comment by jaroslawdabrowski** [comment]
@JudahGabriel how did you manage to make 1-5 work? We have a multi tenant web application for which we would like start develop mobile apps using pwa. Of course we would like to have only one app, not for every tenant. We have the root domain that is being used as a starting point and it can direct users to their instances. This works fine for ios using allowedOrigins, works fine from android using additionalTrustedOrigins, but I cannot make it work for windows. There is no such configuration option for windows and I get the address bar for my domains outside the root domain. I was trying to use url_handlers and scope_extensions in my manifest, but none of that worked out. Can you please help me with this?

**Comment by JudahGabriel** [comment]
@jaroslawdabrowski 

We haven't made that scenario work, because `scope_extensions` are not yet standardized. There may be a trial origin you can enable for your web app to get access to scope_extensions today. I'll email you privately.

**Comment by sertxudev** [comment]
We have a multi-subdomain internal web application (4 businesses with one subdomain each) under a domain that manages the users and sessions.

Example:
- intranet.webapp.test (Login)
- A.intranet.webapp.test (Business A)
- B.intranet.webapp.test (Business B)
- C.intranet.webapp.test (Business C)
- D.intranet.webapp.test (Business D)

We're looking forward to being able to install a PWA for the Login app and redirect our users to each business app without them noticing.

We're hopping `scope_extensions` will solve this problem.

**Comment by jaroslawdabrowski** [comment]
@sertxudev we are trying the same scenario and also hope that scope_extensions will solve our issue in the future.
